File: src/ReadWise.Core/Entities/Article.cs
namespace ReadWise.Core.Entities;

public class Article
{
    public Guid Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string Url { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Author { get; set; }
    public string? Content { get; set; }
    public string? Excerpt { get; set; }
    public string? ImageUrl { get; set; }
    public string? Domain { get; set; }
    public int WordCount { get; set; }
    public int EstimatedReadingTimeMinutes { get; set; }
    public bool IsContentParsed { get; set; }
    public bool IsRead { get; set; }
    public bool IsArchived { get; set; }
    public bool IsFavorite { get; set; }
    public DateTime SavedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ReadAt { get; set; }
    public DateTime? ArchivedAt { get; set; }

    public ICollection<ArticleTag> ArticleTags { get; set; } = [];
}

File: src/web/src/types/article.ts
export interface Tag {
  id: string;
  name: string;
}

export interface ArticleTag {
  articleId: string;
  tagId: string;
  tag: Tag;
}

export interface Article {
  id: string;
  userId: string;
  url: string;
  title: string;
  author?: string;
  content?: string;
  excerpt?: string;
  imageUrl?: string;
  domain?: string;
  wordCount: number;
  estimatedReadingTimeMinutes: number;
  isContentParsed: boolean;
  isRead: boolean;
  isArchived: boolean;
  isFavorite: boolean;
  savedAt: string;
  readAt?: string;
  archivedAt?: string;
  articleTags?: ArticleTag[];
}

export interface CreateArticleRequest {
  url: string;
  title?: string;
}
